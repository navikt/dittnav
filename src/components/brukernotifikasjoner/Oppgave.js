import React from 'react';
import useSikkerhetsnivaa from '../../hooks/useSikkerhetsnivaa';
import LenkepanelMedIkon from '../common/LenkepanelMedIkon';
import { transformTolokalDatoTid } from '../../utils/datoUtils';
import PanelOverskrift from '../common/PanelOverskrift';
import IkonOppgave from '../../assets/IkonOppgave';
import OppgaveType from '../../types/OppgaveType';
import InnloggingsstatusType from '../../types/InnloggingsstatusType';
import { listOfActions, listOfComponentNames } from '../../utils/amplitudeUtils';

const Oppgave = ({ oppgave, innloggingsstatus }) => {
  const sikkerhetsnivaa = useSikkerhetsnivaa(oppgave, 'oppgave', innloggingsstatus);
  const overskrift = <PanelOverskrift overskrift={sikkerhetsnivaa.tekst} type="Element" />;
  const lokalDatoTid = transformTolokalDatoTid(oppgave.eventTidspunkt);

  return (
    <LenkepanelMedIkon
      className="oppgave"
      alt="Oppgave"
      overskrift={overskrift}
      etikett={lokalDatoTid}
      href={sikkerhetsnivaa.lenke}
      amplitudeAction={listOfActions.TrykkPaaBrukernotifikasjon}
      amplitudeComponentName={listOfComponentNames.brukernotifikasjon.OppgaveMottatt}
    >
      <IkonOppgave />
    </LenkepanelMedIkon>
  );
};

Oppgave.propTypes = {
  oppgave: OppgaveType,
  innloggingsstatus: InnloggingsstatusType,
};

Oppgave.defaultProps = {
  oppgave: null,
  innloggingsstatus: null,
};

export default Oppgave;
