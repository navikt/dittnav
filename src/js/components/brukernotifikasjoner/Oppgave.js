import React from 'react';
import { useLocation } from 'react-router-dom';
import useSikkerhetsnivaa from '../../hooks/useSikkerhetsnivaa';
import LenkepanelMedIkon from '../common/LenkepanelMedIkon';
import transformTolokalDatoTid from '../../utils/DatoUtils';
import PanelOverskrift from '../common/PanelOverskrift';
import IkonOppgave from '../../../assets/IkonOppgave';
import OppgaveType from '../../types/OppgaveType';
import InnloggingType from '../../types/InnloggingType';
import { GoogleAnalyticsAction, removeFragment } from '../../utils/GoogleAnalytics';

const Oppgave = ({ oppgave, innlogging }) => {
  const location = useLocation();
  const sikkerhetsnivaa = useSikkerhetsnivaa(oppgave, 'oppgave', innlogging);
  const overskrift = <PanelOverskrift overskrift={sikkerhetsnivaa.tekst} type="Element" />;
  const lokalDatoTid = transformTolokalDatoTid(oppgave.eventTidspunkt);

  return (
    <LenkepanelMedIkon
      className="oppgave"
      alt="Oppgave"
      overskrift={overskrift}
      etikett={lokalDatoTid}
      href={sikkerhetsnivaa.lenke}
      gaCategory={`Ditt NAV${location.pathname}`}
      gaAction={GoogleAnalyticsAction.Oppgave}
      gaUrl={removeFragment(sikkerhetsnivaa.lenke)}
    >
      <IkonOppgave />
    </LenkepanelMedIkon>
  );
};

Oppgave.propTypes = {
  oppgave: OppgaveType,
  innlogging: InnloggingType,
};

Oppgave.defaultProps = {
  oppgave: null,
  innlogging: null,
};

export default Oppgave;
