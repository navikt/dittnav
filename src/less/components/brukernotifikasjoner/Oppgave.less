.oppgave {
  margin-bottom: 1em;
  padding: 0.75em 0.75em !important;
  display: flex;
  flex-direction: row;
  background-color: #fff8ef !important;
  border: 1px solid @navOransjeLighten60 !important;

  .lenkepanel__innhold {
    padding: 0;
  }

  .lenkepanel__ikon {
    align-self: center;

    svg {
      vertical-align: middle;
    }
  }

  &__tekst {
    align-self: center;
    padding-left: 1em;
  }

  &__knapp {
    align-self: center;
    margin-left: auto;
    padding-right: 0.5em;
  }
}

@media screen and (max-width: @mobile-breakpoint) {
  .oppgave {
    padding: 1.035em 1.035em;
    padding-top: 1.25em !important;
    padding-bottom: 1.25em !important;

    .lenkepanel__ikon {
      display: none;
    }
  }
}
