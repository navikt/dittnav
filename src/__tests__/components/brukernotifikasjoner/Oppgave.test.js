import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import Oppgave from 'js/components/brukernotifikasjoner/Oppgave';
import wrapIntl from 'js/IntlTestHelper';

test('Oppgave with sikkerhetsnivaa 4 and innloggingsnivaa 4', () => {
  const oppgave = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Du har en sykemelding som m책 godkjennes',
    link: 'https://enNyLenke',
    produsent: 'dittnav',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Oppgave oppgave={oppgave} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Oppgave with sikkerhetsnivaa 4 and innloggingsnivaa 3', () => {
  const oppgave = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Du har en sykemelding som m책 godkjennes',
    link: 'https://enNyLenke',
    produsent: 'dittnav',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 4,
  };

  const innlogging = {
    securityLevel: '3',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Oppgave oppgave={oppgave} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Oppgave with sikkerhetsnivaa 3 and innloggingsnivaa 3', () => {
  const oppgave = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Du har en sykemelding som m책 godkjennes',
    link: 'https://enNyLenke',
    produsent: 'dittnav',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 3,
  };

  const innlogging = {
    securityLevel: '3',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Oppgave oppgave={oppgave} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Oppgave with sikkerhetsnivaa 3 and innloggingsnivaa 4', () => {
  const oppgave = {
    eventTidspunkt: '2020-03-13T08:53:24.636Z',
    eventId: '1584093204636',
    tekst: 'Du har en sykemelding som m책 godkjennes',
    link: 'https://enNyLenke',
    produsent: 'dittnav',
    sistOppdatert: '2020-03-13T08:53:25.002983Z',
    sikkerhetsnivaa: 3,
  };

  const innlogging = {
    securityLevel: '4',
  };

  const component = ReactTestRenderer.create(wrapIntl(<Oppgave oppgave={oppgave} innlogging={innlogging}/>));
  expect(component.toJSON()).toMatchSnapshot();
});
