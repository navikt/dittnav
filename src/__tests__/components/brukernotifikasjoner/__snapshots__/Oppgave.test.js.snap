// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oppgave with sikkerhetsnivaa 3 and innloggingsnivaa 3 1`] = `
<a
  className="lenkepanel oppgave lenkepanel--border"
  href="https://enNyLenke"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon_test"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#FFC166"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M38.5 39C38.776 39 39 38.776 39 38.5V15.5C39 15.224 38.776 15 38.5 15H15.5C15.224 15 15 15.224 15 15.5V38.5C15 38.776 15.224 39 15.5 39H38.5ZM38 16V38H16V16H38ZM20.5 25C20.372 25 20.244 24.951 20.146 24.854L18.146 22.854C17.951 22.659 17.951 22.342 18.146 22.147C18.341 21.952 18.658 21.952 18.853 22.147L20.5 23.793L25.146 19.147C25.341 18.952 25.658 18.952 25.853 19.147C26.048 19.342 26.048 19.659 25.853 19.854L20.853 24.854C20.756 24.951 20.628 25 20.5 25ZM20.146 33.854C20.244 33.951 20.372 34 20.5 34C20.628 34 20.756 33.951 20.853 33.854L25.853 28.854C26.048 28.659 26.048 28.342 25.853 28.147C25.658 27.952 25.341 27.952 25.146 28.147L20.5 32.793L18.853 31.147C18.658 30.952 18.341 30.952 18.146 31.147C17.951 31.342 17.951 31.659 18.146 31.854L20.146 33.854ZM35.5 24H27.5C27.224 24 27 23.776 27 23.5C27 23.224 27.224 23 27.5 23H35.5C35.776 23 36 23.224 36 23.5C36 23.776 35.776 24 35.5 24ZM27.5 33H35.5C35.776 33 36 32.776 36 32.5C36 32.224 35.776 32 35.5 32H27.5C27.224 32 27 32.224 27 32.5C27 32.776 27.224 33 27.5 33Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Du har en sykemelding som må godkjennes
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;

exports[`Oppgave with sikkerhetsnivaa 3 and innloggingsnivaa 4 1`] = `
<a
  className="lenkepanel oppgave lenkepanel--border"
  href="https://enNyLenke"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon_test"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#FFC166"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M38.5 39C38.776 39 39 38.776 39 38.5V15.5C39 15.224 38.776 15 38.5 15H15.5C15.224 15 15 15.224 15 15.5V38.5C15 38.776 15.224 39 15.5 39H38.5ZM38 16V38H16V16H38ZM20.5 25C20.372 25 20.244 24.951 20.146 24.854L18.146 22.854C17.951 22.659 17.951 22.342 18.146 22.147C18.341 21.952 18.658 21.952 18.853 22.147L20.5 23.793L25.146 19.147C25.341 18.952 25.658 18.952 25.853 19.147C26.048 19.342 26.048 19.659 25.853 19.854L20.853 24.854C20.756 24.951 20.628 25 20.5 25ZM20.146 33.854C20.244 33.951 20.372 34 20.5 34C20.628 34 20.756 33.951 20.853 33.854L25.853 28.854C26.048 28.659 26.048 28.342 25.853 28.147C25.658 27.952 25.341 27.952 25.146 28.147L20.5 32.793L18.853 31.147C18.658 30.952 18.341 30.952 18.146 31.147C17.951 31.342 17.951 31.659 18.146 31.854L20.146 33.854ZM35.5 24H27.5C27.224 24 27 23.776 27 23.5C27 23.224 27.224 23 27.5 23H35.5C35.776 23 36 23.224 36 23.5C36 23.776 35.776 24 35.5 24ZM27.5 33H35.5C35.776 33 36 32.776 36 32.5C36 32.224 35.776 32 35.5 32H27.5C27.224 32 27 32.224 27 32.5C27 32.776 27.224 33 27.5 33Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Du har en sykemelding som må godkjennes
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;

exports[`Oppgave with sikkerhetsnivaa 4 and innloggingsnivaa 3 1`] = `
<a
  className="lenkepanel oppgave lenkepanel--border"
  href="undefined"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon_test"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#FFC166"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M38.5 39C38.776 39 39 38.776 39 38.5V15.5C39 15.224 38.776 15 38.5 15H15.5C15.224 15 15 15.224 15 15.5V38.5C15 38.776 15.224 39 15.5 39H38.5ZM38 16V38H16V16H38ZM20.5 25C20.372 25 20.244 24.951 20.146 24.854L18.146 22.854C17.951 22.659 17.951 22.342 18.146 22.147C18.341 21.952 18.658 21.952 18.853 22.147L20.5 23.793L25.146 19.147C25.341 18.952 25.658 18.952 25.853 19.147C26.048 19.342 26.048 19.659 25.853 19.854L20.853 24.854C20.756 24.951 20.628 25 20.5 25ZM20.146 33.854C20.244 33.951 20.372 34 20.5 34C20.628 34 20.756 33.951 20.853 33.854L25.853 28.854C26.048 28.659 26.048 28.342 25.853 28.147C25.658 27.952 25.341 27.952 25.146 28.147L20.5 32.793L18.853 31.147C18.658 30.952 18.341 30.952 18.146 31.147C17.951 31.342 17.951 31.659 18.146 31.854L20.146 33.854ZM35.5 24H27.5C27.224 24 27 23.776 27 23.5C27 23.224 27.224 23 27.5 23H35.5C35.776 23 36 23.224 36 23.5C36 23.776 35.776 24 35.5 24ZM27.5 33H35.5C35.776 33 36 32.776 36 32.5C36 32.224 35.776 32 35.5 32H27.5C27.224 32 27 32.224 27 32.5C27 32.776 27.224 33 27.5 33Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Du har fått en oppgave, logg inn med høyere sikkerhetsnivå for å se oppgaven.
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;

exports[`Oppgave with sikkerhetsnivaa 4 and innloggingsnivaa 4 1`] = `
<a
  className="lenkepanel oppgave lenkepanel--border"
  href="https://enNyLenke"
  onClick={[Function]}
>
  <div
    className="lenkepanel__innhold"
  >
    <div
      className="lenkepanel__ikon_test"
    >
      <svg
        fill="none"
        height="48"
        viewBox="0 0 54 54"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="27"
          cy="27"
          fill="#FFC166"
          r="27"
        />
        <path
          clipRule="evenodd"
          d="M38.5 39C38.776 39 39 38.776 39 38.5V15.5C39 15.224 38.776 15 38.5 15H15.5C15.224 15 15 15.224 15 15.5V38.5C15 38.776 15.224 39 15.5 39H38.5ZM38 16V38H16V16H38ZM20.5 25C20.372 25 20.244 24.951 20.146 24.854L18.146 22.854C17.951 22.659 17.951 22.342 18.146 22.147C18.341 21.952 18.658 21.952 18.853 22.147L20.5 23.793L25.146 19.147C25.341 18.952 25.658 18.952 25.853 19.147C26.048 19.342 26.048 19.659 25.853 19.854L20.853 24.854C20.756 24.951 20.628 25 20.5 25ZM20.146 33.854C20.244 33.951 20.372 34 20.5 34C20.628 34 20.756 33.951 20.853 33.854L25.853 28.854C26.048 28.659 26.048 28.342 25.853 28.147C25.658 27.952 25.341 27.952 25.146 28.147L20.5 32.793L18.853 31.147C18.658 30.952 18.341 30.952 18.146 31.147C17.951 31.342 17.951 31.659 18.146 31.854L20.146 33.854ZM35.5 24H27.5C27.224 24 27 23.776 27 23.5C27 23.224 27.224 23 27.5 23H35.5C35.776 23 36 23.224 36 23.5C36 23.776 35.776 24 35.5 24ZM27.5 33H35.5C35.776 33 36 32.776 36 32.5C36 32.224 35.776 32 35.5 32H27.5C27.224 32 27 32.224 27 32.5C27 32.776 27.224 33 27.5 33Z"
          fill="#262626"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <p
        className="typo-element lenkepanel__heading"
      >
        Du har en sykemelding som må godkjennes
      </p>
      
      <p
        className="typo-undertekst lenkepanel__etikett"
      >
        13.03.2020 - 09:53
      </p>
    </div>
  </div>
  <span
    className="lenkepanel__indikator"
  />
</a>
`;
